src/Settings/Flavours/Quickest.hs

```diff
@@ -2,13 +2,16 @@ module Settings.Flavours.Quickest (quickestFlavour) where
 
 import Flavour
 import Predicate
+import Oracles.Config.Flag (platformSupportsSharedLibs)
 import {-# SOURCE #-} Settings.Default
 
 quickestFlavour :: Flavour
 quickestFlavour = defaultFlavour
     { name        = "quickest"
     , args        = defaultBuilderArgs <> quickestArgs <> defaultPackageArgs
-    , libraryWays = append [vanilla]
+    , libraryWays = mconcat
+                    [ append [vanilla]
+                    , notStage0 ? platformSupportsSharedLibs ? append [dynamic] ]
```

Comment:
The difference between `quick` and `quickest` in the Make build system seemed too marginal to me, so I thought it would be better to deviate from the existing definition. The `quickest` flavour should be the one that can't be beaten in terms of speed or else why name it this way :)

At the moment, `quickest` in Hadrian is already different, so I suggest not to try to change this in this PR. This discussion deserves a separate issue.